admin/pr_as_branch

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+set -e
+set -u
+
+
+# Usage
+
+usage() {
+  program=$(basename "$0");
+
+  if [ $# != 0 ]; then echo "$@"; echo ""; fi;
+
+  echo "${program}: usage:";
+  echo "    ${program} <pull_request_number> <trac_ticket_number> <branch_name>";
+
+}
+
+
+# Options
+
+while [ $# != 0 ]; do
+  case "$1" in
+    --help)
+      usage;
+      exit 0;
+      ;;
+    --|*) break; ;;
+  esac;
+done;
+
+if [ $# != 3 ]; then
+    usage "Invalid arguments: $*";
+    exit 1;
+fi;
+
+    PR_NUMBER="$1"; shift;
+TICKET_NUMBER="$1"; shift;
+  BRANCH_NAME="$1"; shift;
+
+
+# Do The Right Thing
+
+#repo="https://github.com/twisted/twisted.git";
+repo="git@github.com:twisted/twisted.git";
+
+clone="$(mktemp -d -t twisted)";
+
+git clone --progress "${repo}" "${clone}";
```

Comment:
This is a _super_ expensive way to do this; it will result in ~100M of network traffic on every invocation, which is, for example, more bandwidth than @hawkowl's totalitarian oppressors allow her in a given month.

Possible alternatives: 

- `git clone --depth 1 --progress "${repo}" "${clone}"` will result in an order-of-magnitude bandwidth reduction if you need to re-clone every time for some reason
- check for a `.git` directory in $PWD
- look at `"$(dirname $(dirname "$0"))/.git"` since if they have this script presumably they have a repository checkout already